### PR TITLE
Clean up some unused code

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

Cleans up code based on the result of `unused` check

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

All code is unused

### CODE REVIEW

File: central/sac/authorizer/builtin_scoped_authorizer.go [hunk 1]
```diff
@@ -23,8 +23,6 @@ var (
 	ErrUnexpectedScopeKey = errors.New("unexpected scope key")
 	// ErrUnknownResource is returned when resource is unknown.
 	ErrUnknownResource = errors.New("unknown resource")
-
-	errNoGlobalEffectiveAccessScope = errors.New("global scope checker has no effective access scope")
```
Comment (connorgorman) [inline central/sac/authorizer/builtin_scoped_authorizer.go:27]: @rukletsov @janisz This is fine right?

File: central/scannerdefinitions/handler/handler.go [hunk 2]
```diff
@@ -165,11 +165,6 @@ func writeErrorForFile(w http.ResponseWriter, err error, path string) {
 	httputil.WriteGRPCStyleErrorf(w, codes.Internal, "could not read file %s: %v", filepath.Base(path), err)
 }
 
-func serveFile(w http.ResponseWriter, r *http.Request, name string) {
-	log.Debugf("Serving vulnerability definitions from %s", filepath.Base(name))
-	http.ServeFile(w, r, name)
-}
-
```
Comment (connorgorman) [inline central/scannerdefinitions/handler/handler.go:172]: @RTann This is fine right?

File: pkg/sac/search_helper.go [hunk 3]
```diff
@@ -206,7 +206,6 @@ func (h *searchHelper) filterResults(ctx context.Context, resourceScopeChecker S
 
 type pgSearchHelper struct {
 	resourceMD          permissions.ResourceMetadata
-	optionsMap          search.OptionsMap
```
Comment (connorgorman) [inline pkg/sac/search_helper.go:209]: @janisz @rhybrillou This is expected?

File: central/scannerdefinitions/handler/handler.go (hunk 2 above)
Comment (RTann) [reply central/scannerdefinitions/handler/handler.go:172]: If it's unused I'd say so. @jvdm modified the scannerdefinitions handler recently, so he can probably confirm this, though

**Review by viswajithiii — APPROVED** [review]

**Pull request merged**